app/views/books/index.html.erb

```diff
@@ -24,12 +22,12 @@
     <% @books.each do |book| %>
       <tr>
         <td><%= book.title %></td>
-        <td><%= book.publisher_id %></td>
+        <td><%= book.publisher.name if book.publisher %></td>
         <td><%= book.published_date %></td>
-        <td><%= book.description %></td>
+        <td><% book.description %></td>
         <td><%= book.isbn %></td>
         <td><%= book.page_count %></td>
-        <td><%= book.image_link %></td>
+        <td><%= image_tag book.image_link if book.image_link %></td>
```

Comment:
Just curious what if we removed the `if` condition?

Reply:
It will return an error `image_tag nil`

Reply:
- forget about it, it was a problem and I solved it with format validation for image_link to check if it is link or not look at new commit